app/javascript/crayons/formElements/Toggles/Toggle.jsx

```diff
@@ -0,0 +1,17 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+
+export const Toggle = ({ description, ...otherProps }) => {
+  return (
+    <label class="c-toggle" aria-label={description}>
```

Comment:
@aitchiss can you think of a better way to solve it? If I don't provide aria-label (or actual label), the a11y tools will freak out...

Reply:
Yeah this is the problem with the dev tools + separating the labels from the component 😕  It is always an option to judiciously 'ignore' some issues, but...

Dev tools aside, I think we might need to have a bit more of a think about how we are implementing this toggle - I noticed in the Spaces section we have a `label` inside a `label` to build this control (only one of them containing the actual label info). This can have [really funky consequences for assistive technology](https://web.dev/form-field-multiple-labels/).  I guess it's an unusual case because unlike other form field components, this one is reliant on the label element to make it properly styled and clickable.

I don't think there's any reason to not use the `<label>` in this file as the "one and only" label for this checkbox, other than the desire to separate labels from inputs at a design system level (which is 100% fair), and I'm not quite sure how to resolve it (yet).

If we don't have a label here it's not gonna be functional, but if we do have a label here, it should really be the actual input's label and not have any funky additional labels/label nesting when we use the component, but that's kinda yucky from the design system view.

I'll have a think about it and get back to this review tomorrow morning, I just thought I'd mention it now in case you have any ideas too 😄

Reply:
I generally agree. I don't think we should have `label > label.c-toggle` situation and what we have in Space section is.. wrong. Would the appropriate solution in that particular case be `label + label.c-toggle`?

If yes, would we still need that aria-label (or any sort of description) for `.c-toggle` for a11y or can we ignore this issue?

That being said, I genuinely think THIS component should be built using `label` element because only (?) that gives us the interactivity we want without unnecessary JS (I've seen similar toggles built with `button` but these required JS to switch the state).

But I'd also love need toggle component to not affect/define other things: I don't want it to define HOW the actual text label should look like so this element _should_ not be part of the component..

Reply:
Now I'm also thinking this could also potentially be just a prop flag `areYouInsideLabel` or something which would define what's the outer element (`label` vs `span`) - but I'm not a huge fan of adding such a granular configuration..

Reply:
> That being said, I genuinely think THIS component should be built using label element because only (?) that gives us the interactivity we want without unnecessary JS (I've seen similar toggles built with button but these required JS to switch the state).

Yes, I agree. The other downside to using `button` is it's not a form component, so if we have this toggle as part of an overall form, we'd have to have some kind of hidden input tracking its state, etc 🙃  Let's avoid all those shenanigans. 

> But I'd also love need toggle component to not affect/define other things: I don't want it to define HOW the actual text label should look like so this element should not be part of the component..

Yeah this is the conflict - we want to have this component define the label but also have it not define the visual label... which is kinda messing with semantics and like that article I linked mentions, has negative consequences for screen reader users. 

The main problem with the double labelling is that some screen readers will pick one label to announce, some will pick the other, and some will announce both. This becomes a breaking problem if the reader picks an empty label (like the one inside this component) to announce. It's more of an annoyance in the other cases.

In saying that, `aria-label` _should_ override any `<label>` elements, so I think perhaps all we can do in this case is require a `screenReaderLabel` prop and add it to the `input` with `aria-label={screenReaderLabel}` (I think this should silence the a11y linter warnings too). I don't really like it because it's different to our other form components, and I think there's a high likelihood that developers will forget to add this property, especially if they're creating it in a plain HTML situation (which TBF is most forms in our app). But I can't see a better way round it for now.

If that sounds OK to you, can we update the storybook example so that it demonstrates how to use the toggle with a label, as well as just having the component itself? I think we need to document as much as clearly as possible what's going on with this one 😄

Reply:
Ooh my message crossed with your last one - checking out the codepen now 😄

Reply:
...Also my codepen wouldn't eliminate violation in storybook but assuming it will be used correctly in actual product, I think we could potentially ignore that Storybook error.

Reply:
> 1. I'm not sure if it's bullet-proof enough? i've tested it in Chrome, Safari and FF and it seemed ok...
> 2. I don't know how naughty that checkbox trick is, especially for a11y...

Oooh this is great! It seems to work really well 🤩  Behaved well for me (inc. on touch device), and yeah plays nicely with wrapping it all in a `label` or however dev chooses to label it. I don't see any issues from an a11y POV - at the end of the day it's still a checkbox, it's still keyboard focusable, and as long as we set opacity to `0.00001` (our `--opacity-0`) then it shouldn't be hidden in any way for screen reader users.

This is such a better solution than forcing some weird `aria-label` 😭 🎉

Reply:
> ...Also my codepen wouldn't eliminate violation in storybook but assuming it will be used correctly in actual product, I think we could potentially ignore that Storybook error.

I think if you add a label to the Storybook story then it will be happy (and probably it is nice to showcase how devs should be implementing this component with label?)

Reply:
Okkkey, I think it works now.. Can you give it another look?